src/Dropbox.elm

```diff
@@ -720,9 +729,41 @@ type alias UploadResponse =
     }
 
 
-decodeUploadResponse : Json.Decode.Decoder UploadResponse
-decodeUploadResponse =
-    Pipeline.decode UploadResponse
+{-| Folder metadata
+-}
+type alias FolderMetadata =
+    { name : String
+    , id : String
+    , pathLower : Maybe String
+    , pathDisplay : Maybe String
+    , parentSharedFolderId : Maybe String
+    , sharedFolderId : Maybe String
+    , sharingInfo : Maybe FileSharingInfo
+    , propertyGroups : Maybe (List PropertyGroup)
+    }
+
+
+{-| Deleted item metadata
+-}
+type alias DeletedMetadata =
+    { name : String
+    , pathLower : Maybe String
+    , pathDisplay : Maybe String
+    , parentSharedFolderId : Maybe String
+    }
+
+
+{-| Metadata union type
+-}
+type Metadata
+    = FileMeta FileMetadata
+    | FolderMeta FolderMetadata
+    | DeletedMeta DeletedMetadata
```

Comment:
The `FooMeta FooMetadata` types here are a bit awkard, but `File` and `Folder` are already used in the `Conflict` error type. Options I considered were moving the Metadata or Error types to their own modules, or altering the error case to be more specific, e.g. `FileConflict`